src/main/java/teammates/ui/controller/Action.java

```diff
@@ -576,10 +584,131 @@ private boolean isMasqueradeModeRequested(AccountAttributes loggedInUser, String
                && loggedInUser.googleId != null
                && !loggedInUser.googleId.equals(requestedUserId);
     }
+
+    /**
+     *Common method to Get the feedback data
+     */
+
+    protected FeedbackSessionAttributes extractFeedbackSessionData() {
+
+        FeedbackSessionAttributes newSession = new FeedbackSessionAttributes();
+
+        newSession.setCourseId(getRequestParamValue(Const.ParamsNames.COURSE_ID));
+        newSession.setFeedbackSessionName(Sanitizer.sanitizeTitle(
+                getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_NAME)));
+
+        // getting the class name from which it's being called
+        String className = this.getClass().getName();
+
+        if ("InstructorFeedbackEditSaveAction".equals(className)) {
```

Comment:
@AnaghHegde 
No, that is done automatically due to polymorphism.

If you have
```
class Action {
  protected void doSomething() {
    // do something general
  }
  protected FeedbackSessionAttributes extractFeedbackSessionData() {
    this.doSomething();
  }
}
```
and
```
class InstructorFeedbackEditSaveAction extends Action {
  @Override
  protected void doSomething() {
    // do something specific to this subclass
  }
}
```
, then the subclass version of `doSomething` will automatically be called when the `Action` object, whose `extractFeedbackSessionData` method is being called, is actually an `InstructorFeedbackEditSaveAction` object. No explicit checking of the object type needs to occur, so there is no need to get any classname.

Of course, this is a contrived example, but it should illustrate my point. Think about how the  code that is specific to the `InstructorFeedbackEditSaveAction` can be extracted in this manner.